### ISSUE TITLE
How to implement Anonymous Subscribe?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [BinaryBlue/imsdroid#135]
```
What steps will reproduce the problem?
1.I want to implement Anonymous Subscribe.But I call setFromUri method to 
define "From: Anonymous <sip:anonymous@anonymizer.invalid>".But the message is 
"From: <sip:Anonymous>".

2.The message is as follow:

SUBSCRIBE tel:+491721092755 SIP/2.0
Via: SIP/2.0/UDP 172.30.97.2;branch=z9hG4bK870872969
From: Anonymous <sip:anonymous@anonymizer.invalid>;tag=3255982263
To: <tel:+491721092755>
Call-ID: 1758623042@172.30.97.2
CSeq: 1 SUBSCRIBE
Contact: <sip:172.30.97.2>;transport=udp
P-Preferred-Identity: sip:simba@ims.chinamobile.com
Route: <sip:10.170.32.11:5060;transport=udp;lr>
Max-Forwards: 70
Accept: application/pidf+xml
P-Access-Network-Info: 3GPP-UTRAN-FDD;utran-cell-id-3gpp=1130103843842743
Content-Length: 0
Expires: 0
Event: presence
Privacy: id

3.How to implement it?

What is the expected output? What do you see instead?


What version of the product are you using? On what operating system?
IMSDroid 1.1.343  on Android 2.2

Please provide any additional information below.


```

Original issue reported on code.google.com by `nieyaol...@gmail.com` on 24 Dec 2010 at 2:17

**Comment by GoogleCodeExporter** [comment]
```
setFromUri("Anonymous <sip:anonymous@anonymizer.invalid>");
This is my call.
```

Original comment by `nieyaol...@gmail.com` on 24 Dec 2010 at 2:30

**Comment by GoogleCodeExporter** [comment]
```
Actually,I want to set From header like this:
"From:Anonymous <sip:<sip:anonymous@anonymizer.invalid>"
not "From:<sip:Anonymous>"

```

Original comment by `twbc.cla...@gmail.com` on 29 Dec 2010 at 3:27

**Comment by GoogleCodeExporter** [comment]
```
Actually,I want to set From header like this:
"From:Anonymous <sip:anonymous@anonymizer.invalid>" or 
"From:\"Anonymous\" <sip:anonymous@anonymizer.invalid>"
not "From:<sip:Anonymous>"


```

Original comment by `Splash.P...@gmail.com` on 1 Jan 2011 at 5:11

**Comment by GoogleCodeExporter** [comment]
```
Could you give a code of anonymous subscribe? 
The message as follow:

SUBSCRIBE tel:+491721092755 SIP/2.0
Via: SIP/2.0/UDP 172.30.97.2;branch=z9hG4bK870872969
From: Anonymous <sip:anonymous@anonymizer.invalid>;tag=3255982263
To: <tel:+491721092755>
Call-ID: 1758623042@172.30.97.2
CSeq: 1 SUBSCRIBE
Contact: <sip:172.30.97.2>;transport=udp
P-Preferred-Identity: sip:simba@ims.chinamobile.com
Route: <sip:10.170.32.11:5060;transport=udp;lr>
Max-Forwards: 70
Accept: application/pidf+xml
P-Access-Network-Info: 3GPP-UTRAN-FDD;utran-cell-id-3gpp=1130103843842743
Content-Length: 0
Expires: 0
Event: presence
Privacy: id


```

Original comment by `Splash.P...@gmail.com` on 2 Jan 2011 at 5:17

**Comment by GoogleCodeExporter** [comment]
```
The "setFromUri()" function expect a basic unquoted SIP URI without display 
name.
You should use: setFromUri("sip:anonymous@anonymizer.invalid");
Off course this will skip the display name. I will keep the issue open until 
resolved.
```

Original comment by `boss...@yahoo.fr` on 4 Jan 2011 at 4:02
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Well,I hope I can get a method which may add display name+Uri.
```

Original comment by `twbc.cla...@gmail.com` on 4 Jan 2011 at 4:39

**Comment by GoogleCodeExporter** [comment]
```
Because anonymous subscribe must have a display-name "Anonymous",I hope I can 
get a Java function which may add display name+Uri.
Will you provide this Java function soon？

```

Original comment by `twbc.cla...@gmail.com` on 4 Jan 2011 at 4:57

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Fixed by "imsdroid-2.0.419-preview".
```

Original comment by `boss...@yahoo.fr` on 5 Jun 2011 at 5:22
- Changed state: **Fixed**
